### PR TITLE
#15 - upgrade to distillery

### PR DESCRIPTION
**Pull request by rowpsmo1859**
Changelog
- Changed check-exrm.mk to check-distillery.mk
- Changed `mix release` to have proper distillery prod flag passed

### CODE REVIEW

File: edib/make-app.mk [hunk 1]
```diff
@@ -27,7 +27,7 @@ postinfo:
 release: $(RELEASE_FILE)
 
 $(RELEASE_FILE): app-compile phoenix-assets
-	$(IN_APP_DIR) MIX_ENV=$(MIX_ENV) mix release
+	$(IN_APP_DIR) mix release --env=prod
```
Comment (asaaki) [inline edib/make-app.mk]: We probably still need to support custom provided envs, therefore it should be changed to:

```
$(IN_APP_DIR) mix release --env=$(MIX_ENV:-prod)
```

For the future we can think about dividing the mix and the release env, but for now we should stick to what user might expect currently.

WDYT?

File: tools/app_info.exs [hunk 2]
```diff
@@ -1,7 +1,7 @@
 defmodule AppInfo do
   def name,    do: config |> Dict.get(:app)
   def version, do: config |> Dict.get(:version)
-  def exrm,    do: deps |> Dict.has_key?(:exrm)
+  def distillery,    do: deps |> Dict.has_key?(:distillery)
```
Comment (asaaki) [inline tools/app_info.exs]: We can get rid of all the superfluous whitespaces if you like (also from the surrounding functions.
I had the habit of `do` alignments in the past, but won't stick and enforce it anymore.

**Review by asaaki — CHANGES_REQUESTED** [review]

File: edib/make-app.mk (hunk 1 above)
Comment (rowpsmo1859) [reply edib/make-app.mk]: I think that's a good idea. It'll prevent breaking in the scenario you described. I'll push the change shortly

File: tools/app_info.exs (hunk 2 above)
Comment (rowpsmo1859) [reply tools/app_info.exs]: xD I'll try it out for giggles

**Comment by rowpsmo1859** [comment]
Running tests on my changes

**Comment by rowpsmo1859** [comment]
Everything passes. @asaaki Could you review the changes please?

File: edib/make-app.mk [hunk 3]
```diff
@@ -27,7 +27,7 @@ postinfo:
 release: $(RELEASE_FILE)
 
 $(RELEASE_FILE): app-compile phoenix-assets
-	$(IN_APP_DIR) mix release --env=$(MIX_ENV:-prod)
```
Comment (asaaki) [inline edib/make-app.mk]: Ah, sorry, actually my bad.

What I meant was `--env=${MIX_ENV:-prod}` which would work in a shell session. But in make's case we already have a default value if not set.
(So, good spot!)

**Review by asaaki — APPROVED** [review]

**Comment by asaaki** [comment]
Everything looks good.
Thank you for your work! 🙇

**Pull request merged**